docs/design-notes/wasm.md

```diff
@@ -112,42 +110,42 @@ for memory management.
 Example source code:
 
 ```c
-struct __attribute__((packed)) nullable_bigint { 
-    int size;
-    long long v;
-};
+const int WASM_PAGE_SIZE = 64 * 1024;
+
+const int _scylla_abi = 0;
 
-static long long swap_int64(long long val) { 
+static long long swap_int64(long long val) {
     val = ((val << 8) & 0xFF00FF00FF00FF00ULL ) | ((val >> 8) & 0x00FF00FF00FF00FFULL );
     val = ((val << 16) & 0xFFFF0000FFFF0000ULL ) | ((val >> 16) & 0x0000FFFF0000FFFFULL );
     return (val << 32) | ((val >> 32) & 0xFFFFFFFFULL);
 }
 
-long long fib_aux(long long n) { 
-    if (n < 2) { 
+long long fib_aux(long long n) {
+    if (n < 2) {
         return n;
-    } 
+    }
     return fib_aux(n-1) + fib_aux(n-2);
 }
 
-struct nullable_bigint* fib(struct nullable_bigint* p) { 
-    // Initialize memory for the return struct
-    struct nullable_bigint* ret = (struct nullable_bigint*)__builtin_wasm_memory_size(0);
-    __builtin_wasm_memory_grow(0, sizeof(struct nullable_bigint));
-
-    ret->size = sizeof(long long);
-    if (p->size == -1) { 
-        ret->v = swap_int64(42);
-    } else { 
-        ret->v = swap_int64(fib_aux(swap_int64(p->v)));
-    } 
-    return ret;
+long long fib(long long p) {
+    int size = p >> 32;
+    long long* p_val = (long long*)(p & 0xffffffff);
+    // Initialize memory for the return value
+    long long* ret_val = (long long*)(__builtin_wasm_memory_size(0) * WASM_PAGE_SIZE);
+    __builtin_wasm_memory_grow(0, 1); // long long fits in one wasm page
```

Comment:
I thought we switched to malloc/free.

Reply:
There was a problem with switching in test_wasm.py but I think I can change it here